src/controllers/controller.doughnut.js

```diff
@@ -78,6 +78,35 @@ export default class DoughnutController extends DatasetController {
 		return ringIndex;
 	}
 
+	/**
+	 * Get the maximal rotation & circumference extents
+	 * across all visible datasets.
+	 */
+	_getRotationExtents() {
```

Comment:
We don't really have a better spot for this. It is O(n^2) on the number of datasets since each dataset runs this code. Perhaps we need to factor out common setup like this. Similar runtime characteristics exist for the doughnut weights

Reply:
I'd be approving that change :)